accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -933,7 +937,17 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
                 continue;
             }
 
-            possible_evictions.insert(0, *k, Arc::clone(v));
+            // Skip entries with ref_count != 1 early
+            // In 99% of cases, these will be rejected by should_evict_from_mem or evict_from_cache anyway
+            // Filtering here avoids unnecessary work and reduces write lock contention in evict_from_cache
+            if v.ref_count() != 1 {
+                continue;
+            }
+
+            // Capture dirty flag early during scan
+            let is_dirty = v.dirty();
+
+            possible_evictions.insert(0, *k, is_dirty);
```

Comment:
IMO the comment and local variable aren't useful.

```suggestion
            possible_evictions.insert(0, *k, v.dirty());
```